### PR TITLE
Serve statically

### PR DESCRIPTION
**Pull request by usirin**
- [x] Client server is being served on `http://localhost:9001` on dev, `https://rope.live` on prod
- [x] Api server is being served on `http://localhost:9000` on dev, `https://api.rope.live` on prod /cc @cihangir @gokmen 
- [x] server is being served via `npm run watch:server` which triggers `gulp watch-server`
- [x] client is being served via `index.html`
- [x] client checks if user is logged in via `GET API_URL/whoami`
- [x] `index.ejs` is removed
- [x] Readme.md is updated with the new commands for development
- [x] Logout works correctly
- [x] `npm run watch:client` starts watching client
- [ ] Create a `Dockerfile` for static server (probably an nginx container)

I am guessing we are gonna serve the static/client server via an nginx container but right now, on dev just using `gulp` will activate client/static server.

basically if somebody wants to start developing on this repo they will have to run 2 commands now:

```bash
docker-compose up # for running api-server and local mongo
API_URL=http://localhost:9000 gulp # for starting development on client
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -6,16 +6,19 @@ services:
...
       - "9000:9000"
     links:
       - mongo
     volumes:
-      - ./app:/src/app
-      - ./server:/src/server
+      - ./app:/opt/rope/app
+      - ./server:/opt/rope/server
+    depends_on:
+      - mongo
```
Comment (usirin) [inline docker-compose.yml:21]: @cihangir even though i set this here, the server still runs before the `mongo` container on the initial `docker-compose up`, should we use a script to wait for mongo to be available? if so can you point me to an example?

**Comment by usirin** [comment]
@gokmen Only remaining part is to decide how to deploy this to production.

File: docker-compose.yml (hunk 1 above)
Comment (cihangir) [reply docker-compose.yml:21]: There is [this](https://github.com/Eficode/wait-for) script which is used widely.

You can see the example here ```command: sh -c './wait-for db:5432 -- npm start```

**Comment by cihangir** [comment]
https://github.com/ropelive/home/pull/2#issuecomment-345358316

This is already done [here](https://github.com/koding/rope-infra/tree/master/kubernetes#home)

**Review by gokmen — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -19,16 +19,19 @@
     }
   ],
   "scripts": {
-    "start": "gulp",
+    "watch:server": "gulp watch-server",
+    "watch:client": "API_URL=http://localhost:9000 gulp",
```
Comment (cihangir) [inline package.json]: we dont need this configuration steps in package.json. this is not the place for configuration

File: package.json [hunk 3]
```diff
@@ -19,16 +19,19 @@
     }
   ],
   "scripts": {
-    "start": "gulp",
+    "watch:server": "gulp watch-server",
```
Comment (cihangir) [inline package.json]: could we have a unified commands/scripts among our projects?

i have tried to make them unified but we are changing them time to time

cc/ @gokmen

File: package.json (hunk 2 above)
Comment (gokmen) [reply package.json]: I think it was a shortcut for developers instead of configuration. But on the other hand I'm against to using prod values as default, we should have the development configuration as default and we should change/update them while deploy.

File: package.json (hunk 3 above)
Comment (gokmen) [reply package.json]: Agreed, it's better to keep with standards `npm start` should remain as is if needed updated to `gulp watch-server`

**Comment by gokmen** [comment]
@usirin this looks ok to me, except the issues that @cihangir pointed out, for Docker container to serve static page we can simply use following;

```
FROM nginx:alpine

ADD dist /usr/share/nginx/html
```

where `dist` the folder of output after `gulp production` which uglyfies the output as you know.

File: package.json (hunk 2 above)
Comment (usirin) [reply package.json]: > I think it was a shortcut for developers instead of configuration.

Exactly.

> But on the other hand I'm against to using prod values as default.

I read somewhere that having production builds as the default, and then changing the variables depending on the environment -this time for `development`- is the best practice, so i am used to do so. But I agree that this creates confusion, I will make the changes accordingly. :+1:

File: package.json (hunk 3 above)
Comment (usirin) [reply package.json]: The main reason for me to change this to `watch:server` is that this repository will create 2 different servers, 1 for api/backend (nodejs), and 1 for serving client (nginx). I was trying to be explicit.

But since we only need 1 npm script only for starting to server, i am changing this back to `npm  start`.

**Comment by usirin** [comment]
https://github.com/ropelive/home/pull/2#issuecomment-345359000

@cihangir This will only deploy the `nodejs` container right? I am guessing we will need some kind of deployment definition for `nginx` container to serve the static pages. That's why I asked.

**Comment by usirin** [comment]
https://github.com/ropelive/home/pull/2#issuecomment-345400203
> for Docker container to serve static page we can simply use following;
```Dockerfile
FROM nginx:alpine

ADD dist /usr/share/nginx/html
```

@gokmen @cihangir one thing to note that, above `Dockerfile` assumes that `dist` folder is created before this container is built. But since all the builds are happening on different containers, serving with `nginx` may require some changes in the build configuration as well. Not sure about the specifics, just wanted to point out beforehand.

**Comment by usirin** [comment]
The simplest solution comes to my mind is to use (create if necessary) a container with combination of `mhart/alpine-node` and `nginx:alpine` to build and serve at the same time, but i am not the most knowledgeable person when it comes to containers, though.
